src/components/SRNEntry.vue

```diff
@@ -247,10 +248,29 @@ export default {
       this.validateCount = userValidationResponse.validateCount;
       this.invalidLoginMessage = userValidationResponse.invalidLoginMessage;
       this.isLoading = false;
+
+      //clear the input field if entry is incorrect
+      if(this.invalidLoginMessage != ""){
+        this.getLatestEntry["userID"] = "";
+      }
+
+      //incorrect entry the first time
+      if(!this.isUserValid && this.validateCount == 1){
+        var purposeParams = "incorrect-entry"
+        var tempUserIDList = [{userID: userID.toString(), valid: this.isUserValid}]
+        sendSQSMessage(
+            this.purpose,
+            purposeParams,
+            this.redirectTo,
+            this.redirectID,
+            tempUserIDList,
+            authType
+          );
+      }
```

Comment:
I think this is not the right place for this logic.
You need to run this piece of logic when someone either submits (after entering a wrong srn) or clicks "add another srn" (after entering a wrong srn).

2 reasons why this logic should not be here.
- This method is called `authenticateSRN` and it's job should just to be do the authentication and setting the respective variables. After setting the needed variables, it should not care about what to do next.
- You need the above logic to run in two cases - clicking submit and clicking add srn button. So this new logic should be present in those two functions `processForm` and `addField`. And to write the same logic two times in two functions would be redundant, you should make a separate function, something on the lines of, `handleIncorrectEntry()` and use that where you need to.

I hope it was clear enough? Let me know if you have any doubts

Reply:
Actually, `authenticateSRN` is being run in both the logic, so this piece of code also will run both times. But yeah, it makes sense that this method shouldn't worry about what is being done to the result.